### ISSUE TITLE
Changing saturation/regen values

### ISSUE THREAD

**Issue by FatalBlade** [squeek502/AppleSkin#8]
Is there any way for it to detect that the food may not be healing what's at the base level? 

I tried to open basicfoodvalues.class but I don't have the decompliler to fix that information.

**Comment by squeek502** [comment]
Not sure what you mean. What exactly are you trying to do?

**Comment by FatalBlade** [comment]
Alright so basically the issue I was having in the first place was that my players on my server have up to 30 hearts(60hp) total and it was a huge hassle to try and get back to full hp with food. So I went out and found a plugin that would allow me to alter how much food/hunger/saturation each food type would give a player.

I really like the way AppleSkin shows how much you're going to get from the food in the bar and in the inventory but after altering these values it's no longer accurate I was wondering if there was a way to change the config so that you can set it to show what they would actually be.

**Comment by squeek502** [comment]
The plugin would need to be compatible with how AppleSkin gets the food values, which are:

- The `ItemFood.getHealAmount` and `ItemFood.getSaturationModifier` functions
- If [AppleCore](https://github.com/squeek502/AppleCore) is installed, the values gotten from [`FoodEvent.GetPlayerFoodValues`](https://github.com/squeek502/AppleCore/blob/HEAD/java/squeek/applecore/api/food/FoodEvent.java)

Which plugin are you using? Depending on how it works, I might be able to add support for it.

**Comment by FatalBlade** [comment]
HealingFood: https://dev.bukkit.org/projects/healingfood

Hmmm okay I'm not really sure how it does it but if you can add support that would be amazing thank you! 

While I'm here I know it's unrelated but I did want to double check that I did it right. To add this to the server I just drop the same file into the server plugins folder that I would add to my client right?

**Comment by squeek502** [comment]
What version of Minecraft are you using? AppleSkin is not a plugin, it's a Forge mod, so I'm assuming you're using something like Sponge?

**Comment by FatalBlade** [comment]
I'm using Spigot, for the client I put it into MultiMc which does forge/liteloader stuff for you.

**Comment by squeek502** [comment]
In that case, I'm not sure there's a way that this can work in the way that I was imagining (I guess other than providing hard overrides on the client-side, which might be what you were asking for originally?).

In order to get the modified values from the server, AppleSkin would need to be running on the server, but for that to happen the server needs to be able to load Forge mods, so you'd need to run something like Forge or Sponge on the server-side (and if you did that, then you might as well just use something like Hunger Overhaul to modify the food values, since that's already compatible with AppleCore/AppleSkin).

I could potentially add a way to override food values client-side, if that would work for you (you'd then have to make sure that everyone that connects to your server has the correct client-side config for those food values [and that those values match the ones set on the server], though).

**Comment by FatalBlade** [comment]
Yeah that's basically what I was looking for and I don't have a huge server so I think that's doable.  If you wouldn't mind that would be awesome, I really appreciate how helpful you have been overall.

**Comment by squeek502** [comment]
No problem. If you wouldn't mind, could you post a small portion of the food value settings of the Healing Food config so I can see how that's structured?

EDIT: [is it a .yml file like this?](https://github.com/D3ath5643/HealingFood/blob/master/src/config.yml)

**Comment by FatalBlade** [comment]
Exactly like that actually, those are the default values.
